projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -161,12 +164,41 @@ public function update_submenus( $slug, $submenus_to_update ) {
 			return;
 		}
 
-		foreach ( $submenu[ $slug ] as $i => $submenu_item ) {
-			if ( array_key_exists( $submenu_item[2], $submenus_to_update ) ) {
-				$submenu_item[2] = $submenus_to_update[ $submenu_item[2] ];
-				// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
-				$submenu[ $slug ][ $i ] = $submenu_item;
+		/**
+		 * Iterate over all submenu items and add the hide the submenus with CSS classes.
+		 * This is done separately of the second foreach because the position of the submenu might change.
+		 */
+		foreach ( $submenu[ $slug ] as $index => $item ) {
+			if ( ! array_key_exists( $item[2], $submenus_to_update ) ) {
+				continue;
+			}
+
+			$css_classes = empty( $item[4] ) ? 'hide-if-js hide-if-no-js' : $item[4] . ' hide-if-js hide-if-no-js';
+
+			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			$submenu [ $slug ][ $index ][4] = $css_classes;
+		}
+
+		$submenu_items = array_values( $submenu[ $slug ] );
```

Comment:
Instead of cloning all `$submenu[ $slug ]`, can we `$submenu_items[] = $item` in the above loop?
This would also remove the need for checking keys in the loop below.
```
if ( ! array_key_exists( $submenu_item[2], $submenus_to_update ) ) {
  continue;
}
```

Reply:
Yeah, I think we could, but afaik array_key_exists has O(1) so it should be pretty fast. Hmmm.. I don't think there's any considerable impact either way: the array would be bigger while appending elements one by one still has a bit of overhead. 

I don't have a preference :); if you think that appending the elements would make more sense/looks cleaner, we could do that. :)

Reply:
> Yeah, I think we could, but afaik array_key_exists has O(1) so it should be pretty fast.
TIL! 

Let's leave it as is, since removing the one loop shouldn't affect the other.